TcgPocket/Features/Users/Commands/UpdatePasswordCommand.cs

```diff
@@ -0,0 +1,66 @@
+﻿using AutoMapper;
+using FluentValidation;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using System.Text.Json.Serialization;
+using TcgPocket.Shared;
+
+namespace TcgPocket.Features.Users.Commands;
+
+public class UpdatePasswordCommand : IRequest<Response>
+{
+    [JsonIgnore]
+    public string UserName { get; set; }
+    public string CurrentPassword { get; set; }
+    public string NewPassword { get; set; }
+    public string NewPasswordConfirmation { get; set; }
+}
+
+public class UpdatePasswordCommandHandler : IRequestHandler<UpdatePasswordCommand, Response>
+{
+    private readonly IMapper _mapper;
+    private readonly IValidator<UpdatePasswordCommand> _validator;
+    private readonly UserManager<User> _userManager;
+
+    public UpdatePasswordCommandHandler(IValidator<UpdatePasswordCommand> validator, UserManager<User> userManager, IMapper mapper)
+    {
+        _mapper = mapper;
+        _validator = validator;
+        _userManager = userManager;
+    }
+
+    public async Task<Response> Handle(UpdatePasswordCommand command, CancellationToken cancellationToken)
+    {
+        var validationResult = await _validator.ValidateAsync(command, cancellationToken);
+
+        if (!validationResult.IsValid)
+        {
+            var errors = _mapper.Map<List<Error>>(validationResult.Errors);
+            return new Response { Errors = errors };
+        }
+
+        var user = _userManager.Users.FirstOrDefault(x => x.UserName.Equals(command.UserName));
```

Comment:
This avoids potential null reference exceptions and is functionally the same.

```suggestion
        var user = _userManager.Users.FirstOrDefault(x => x.UserName == command.UserName);
```